### PR TITLE
SFX bug fixed

### PR DESCRIPTION
**Pull request by Gursaanj**
Leaving/Restarting a match during a powerup usage will no longer cause the powerup to play forever!

### CODE REVIEW

File: Assets/_PartyParkade/Script/Game/Car/CarAudioSystem.cs [hunk 1]
```diff
@@ -53,6 +56,14 @@ private void SetEngineSound(PerigonSoundEffect engine)
             m_wheelVolume = MinWheelVolume;
         }
 
+        private void ResetEngineSound()
```
Comment (Gursaanj) [inline Assets/_PartyParkade/Script/Game/Car/CarAudioSystem.cs:59]: So it turns out that , SoundManager.StopSFX() being invoked on pregame made engine lose its reference to the audiosource  so I had to cache the reference of engine here and reapply it onPregame.invoke

**Comment by Gursaanj** [comment]
[Powerup SFX can be heard on loop forever](https://trello.com/c/PQVMBXQx/890-powerup-sfx-can-be-heard-on-loop-forever)

**Review by mikeslgoh — APPROVED** [review]
thanks for cleaning up the audio manager

File: Assets/_PartyParkade/Script/Game/Car/CarAudioSystem.cs [hunk 2]
```diff
@@ -28,6 +28,7 @@ public class CarAudioSystem : MonoBehaviour
         private const float MinWheelVolume = 0.3f;
         private float m_wheelSpeed;
         private float m_wheelVolume;
+        private PerigonSoundEffect engineSoundEffect = PerigonSoundEffect.NoSFX;
```
Comment (m4rtint) [inline Assets/_PartyParkade/Script/Game/Car/CarAudioSystem.cs:31]: Don't need to cache this

**Review by m4rtint — APPROVED** [review]

**Pull request merged**